source/dub/dependency.d

```diff
@@ -662,19 +672,20 @@ struct Version {
 		m_version = vers;
 	}
 
+	// skips validation for efficiency
+	private this(string vers, bool _trusted)
+	{
+		m_version = vers;
+	}
+
 	/** Constructs a new `Version` from its string representation.
 
 		This method is equivalent to calling the constructor and is used as an
 		endpoint for the serialization framework.
 	*/
 	static Version fromString(string vers) { return Version(vers); }
 
-	bool opEquals(const Version oth) const {
-		if (isUnknown || oth.isUnknown) {
-			throw new Exception("Can't compare unknown versions! (this: %s, other: %s)".format(this, oth));
-		}
-		return opCmp(oth) == 0;
-	}
+	bool opEquals(const Version oth) const { return opCmp(oth) == 0; }
```

Comment:
I assume unknown is caught at an upper level now ?

Reply:
Yes, turned out that `opCmp` has the exact same lines at the top of its body.